torch/utils/data/dataloader.py

```diff
@@ -187,9 +192,13 @@ def _next_indices(self):
     def _put_indices(self):
         assert self.batches_outstanding < 2 * self.num_workers
         if self.samples_remaining > 0:
-            self.index_queue.put((self.send_idx, self._next_indices()))
-            self.batches_outstanding += 1
-            self.send_idx += 1
+            if self.samples_remaining < self.batch_size and self.drop_last:
+                self._shutdown_workers()
+                raise StopIteration
```

Comment:
That's not a good way to solve this. Call `self._next_indices()` to empty the queue and deplete the sampler's stream, and discard the value. That should be enough to skip that batch.